src/main/java/org/springframework/guice/module/SpringModule.java

```diff
@@ -94,18 +139,55 @@ private void bindConditionally(Binder binder, String name, Class<Object> type, P
 		// But allow binding to named beans
 		binder.withSource("spring-guice").bind(type).annotatedWith(Names.named(name)).toProvider(namedProvider);
 	}
+	
+	private final class PartiallyRefreshableApplicationContext extends AnnotationConfigApplicationContext {
+
+		private final AtomicBoolean partiallyRefreshed = new AtomicBoolean(false);
+
+		/*
+		 * Initializes beanFactoryPostProcessors only to ensure that all
+		 * BeanDefinition's are available
+		 */
+		void partialRefresh() {
+			invokeBeanFactoryPostProcessors(getBeanFactory());
```

Comment:
Not happy about this approach either, but I couldn't determine any other way to get all bean definitions without doing a full refresh.